From 42951d5f1120db1504994b3d15714f14304ba74c Mon Sep 17 00:00:00 2001
From: Steve Dickson <steved@redhat.com>
Date: Mon, 22 Jan 2024 13:23:57 -0500
Subject: [PATCH 04/10] reexport.c: Some Distros need the following include to
 avoid the following error
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

reexport.c: In function ‘connect_fsid_service’:
reexport.c:41:28: error: implicit declaration of function ‘offsetof’ [-Werror=implicit-function-declaration]
   41 |                 addr_len = offsetof(struct sockaddr_un, sun_path) + strlen(addr.sun_path);
      |                            ^~~~~~~~
reexport.c:19:1: note: ‘offsetof’ is defined in header ‘<stddef.h>’; did you forget to ‘#include <stddef.h>’?
   18 | #include "xlog.h"
  +++ |+#include <stddef.h>
   19 |
reexport.c:41:37: error: expected expression before ‘struct’
   41 |                 addr_len = offsetof(struct sockaddr_un, sun_path) + strlen(addr.sun_path);
      |                                     ^~~~~~
cc1: some warnings being treated as errors

Signed-off-by: Steve Dickson <steved@redhat.com>
---
 support/reexport/reexport.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/support/reexport/reexport.c b/support/reexport/reexport.c
index c7bff6a3..1febf598 100644
--- a/support/reexport/reexport.c
+++ b/support/reexport/reexport.c
@@ -9,6 +9,7 @@
 #include <sys/vfs.h>
 #include <unistd.h>
 #include <errno.h>
+#include <stddef.h>
 
 #include "nfsd_path.h"
 #include "conffile.h"
-- 
2.45.2

